class/defaults.yml

```
@@ -7,3 +7,7 @@ parameters:
     http_proxy: ""
     https_proxy: ""
     no_proxy: ""
+    solvers:
+      - http01:
+          ingress:
+            class: 'nginx'
```

Having parameters as arrays makes it very difficult to edit existing entries in the config hierarchy. It might make more sense to have something like 

```suggestion
    solvers:
      nginx_http01:
        http01:
          ingress:
            class: 'nginx'
```

and change the code in `component/main.jsonnet` to

```
solvers: [
  params.solvers[s]
  for s in std.objectFields(params.solvers)
],